servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DnsServiceDiscovererBuilderProvider.java

```diff
@@ -21,5 +21,18 @@
 @FunctionalInterface
 public interface DnsServiceDiscovererBuilderProvider {
 
+    /**
+     * Returns a {@link DnsServiceDiscovererBuilder} based on the (unique) id and
+     * pre-initialized {@link DnsServiceDiscovererBuilder}.
+     * <p>
+     * This method may return the pre-initialized {@code builder} as-is, or apply custom builder settings before
+     * returning it, or wrap it ({@link DelegatingDnsServiceDiscovererBuilder} may be helpful).
+     *
+     * @param id a (unique) identifier used to identify the underlying {@link DnsClient}.
```

Comment:
`DnsClient` is a pkg-private interface. Consider referencing `ServiceDiscoverer`